src/Board.cpp

```diff
@@ -21,9 +22,12 @@ void Board::register_tank() {
   tower_texture.setSmooth(true);
   auto& shot_texture = mStore.get_texture("shotOrange.png");
   shot_texture.setSmooth(true);
-  auto tank = Tank(WIDTH / 2.0f, 50.0f, body_texture, tower_texture, shot_texture);
+  auto tank = Tank(WIDTH / 2.0f, 50.0f, body_texture, tower_texture, shot_texture,
+                   std::make_unique<SquareRootEngine>(70, 5));
   tank.set_rotation(180);
-  mTanks.push_back(std::move(tank));
+  mTanks.emplace_back(std::move(tank));
```

Comment:
You don't need emplace_back, it would do the same.

You can keep as it is, but please remember that `emplace_back` is usefull only when you construct objects in-place. Above code will do the same what `push_back` does.

Reply:
Thanks for this explanation